Client/Frontend/ContentBlocker/ContentBlockerHelper.swift

```diff
@@ -0,0 +1,316 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import WebKit
+import Shared
+
+fileprivate let NotificationContentBlockerReloadNeeded = "NotificationContentBlockerReloadNeeded"
+
+@available(iOS 11.0, *)
+class ContentBlockerHelper {
+    static let PrefKeyEnabledState = "prefkey.trackingprotection.enabled"
+    static let PrefKeyStrength = "prefkey.trackingprotection.strength"
+    fileprivate let blocklistBasic = ["disconnect-advertising", "disconnect-analytics", "disconnect-social"]
+    fileprivate let blocklistStrict = ["disconnect-content", "web-fonts"]
+    fileprivate let ruleStore: WKContentRuleListStore?
+    fileprivate weak var tab: Tab?
+    fileprivate weak var profile: Profile?
+
+    // Raw values are stored to prefs, be careful changing them.
+    enum EnabledOption: String {
+        case on = "option-on"
```

Comment:
It will, I didn't know if this would make it clearer it also gets used as pref storage option, and if someone needed to rename the enum they could do so independently of the stored pref value. I see elsewhere in the code we don't do this, so I'll remove the explicit string